lib/transaction.js

```diff
@@ -14,35 +14,12 @@ const { uniqueId, isUndefined } = require('lodash');
 // Acts as a facade for a Promise, keeping the internal state
 // and managing any child transactions.
 class Transaction extends EventEmitter {
-  constructor(client, container, config, outerTx) {
+  constructor(client, container, config = {}, outerTx = null) {
     super();
+    this.doNotRejectOnRollback = config.doNotRejectOnRollback;
```

Comment:
@kibertoad : Should we provide an ULTRA-default value here?  If so: what should it be?

Reply:
I would still default to true, TBH. This whole parameter exists to preserve compatibility with old-style (callback) transactions that used to work the "FALSE" way, and personally I find this behaviour less helpful (see https://github.com/knex/knex/issues/3246)

Reply:
k -- just pushed an update for that.

Reply:
On a related note: perhaps the behavior can eventually be extracted out of the Transactor class via a Decorator or Proxy of some kind?

Reply:
IIRC, Proxy are rather expensive performance-wise, so I would prefer to avoid them on a hot path. Simple decorator might make sense, though.